controller/aks-cluster-config-handler.go

```diff
@@ -849,6 +853,10 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, config *aksv1.
 	}
 
 	if updateAksCluster {
+		// If status is not updating, then enqueue the update ( to re-enter the onChange handler )
+		if config.Status.Phase != aksConfigUpdatingPhase {
+			return h.enqueueUpdate(config)
+		}
```

Comment:
Technically, this check can be moved down to line 866 where the resource group is created if it doesn't exist, no? Since until that point, no other change is being made that would require the status to be set to `Updating`.

Reply:
sure. I have moved the code. Thanks